### DISCUSSION TITLE
ImportError: can't import name numpy

### DISCUSSION THREAD

**Discussion by NibaliDev** [micropython/micropython#15581] [ESP32]
I recently flashed new micropython formware with ulab included. When I then try to import numpy from ulab, I get an ImportError. But whats most weird, is that I don't get an ImportError when only importing ulab. My conclusion is that my instance of ulab don't work according to the official documenation. 

This is the .bin file I flashed on my ESP32: https://gitlab.com/rcolistete/micropython-samples/-/blob/master/ESP32/Firmware/v1.12_with_ulab/ulab_v0.54.0_2020-07-29/Generic_flash-4MB/esp32_idf3_ulab_dp_thread_v1.12-663-gea4670d5a_2020-07-29.bin

**Comment by davefes** [comment]
That image is probably a couple of years old.  Have you searched this forum and the old forum for any threads about using this image?

And specifically, which ESP32 are you using?

**Reply by davefes** [reply]
Seen this:
https://github.com/v923z/micropython-ulab

**Reply by NibaliDev** [reply]
I have an ESP32-WROOM-32. I only found one discussion about the image I'm using, wich is the same diuscussion where I found the image from the very beginning. There I can't really find anything on how to use this image, and I think you're theory of it being too old is totaly right. 

Why I wanted to download a pre-built .bin file, and not build the firmware myself, is totally because i find it much easier. I would therefore be very glad if someone could find another .bin file, that is not outdated. I have tried to build the firmware myself with no succed; maybe you could explain something I'm missing, don't you only need to run the script, or is it much harder than that? 

(I'm sorry if this is a rookie question, but I'm still very new to MicroPython development, and only want ulab to work as fast as possible because of a big project I'm planning).

**Comment by davefes** [comment]
There is a link:
https://github.com/v923z/micropython-builder
that appears to make the process easier.

A year or ago I tried building my own ESP32 images and did not find it easy.  Better tutorials are around now, but I haven't had the need to try again.

Maybe a direct question in the new thread asking for a `.bin` file for the ESP32 Generic V1.24 might do it.
